### ISSUE TITLE
Extract payment status handling

### ISSUE THREAD

**Issue by elia** [solidusio-contrib/solidus_stripe#160]
## What

The code is living in the intents controller and should be extracted to a place in which it can be shared with webhooks.

Some consideration should be given to users installing the extension and needing to customize it in its reaction to payment status changes (what should live in the extension? what should be copied in the target rails app?).

### Dependencies
- blocked by #154 

## Why

The payment status handling is basically the same when returning from a payment confirmation and when receiving webhooks, so we should have a separate helper object that holds the shared behavior.

**Comment by elia** [comment]
Assuming using the extracted service we'll get updates from webhooks as well landing on the return url provided to Stripe (intents controller) we should check if there a way to ensure the information is still fresh. 

E.g. a webhook update comes late, after we already completed the order, with a payment intent status that maps to "processing" thus generating an invalid payment state machine transition. 

Can we ensure that is discarded as stale?

@waiting-for-dev dumping this consideration here, but please let me know if you thing it's better to address it in a separate ticket.
- [ ] Is there's a way to track i know if a payment update is stale, given they can come also from webhooks and have latency)

**Comment by waiting-for-dev** [comment]
Closing in favor of #179, #180, #181, #184, #188 & #189.
